File: Ai-Service/app/models/document.py
from sqlalchemy import Column, String, Integer, DateTime, Text, BigInteger
from sqlalchemy.sql import func
from app.db import Base

class Document(Base):
    __tablename__ = "documents"

    id = Column(Integer, primary_key=True, autoincrement=True, index=True)
    doc_id = Column(String(255), nullable=False, unique=True, index=True)
    user_id = Column(String(255), nullable=False, index=True)
    name = Column(String(255), nullable=True)
    chunk_count = Column(Integer, nullable=False, default=0)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), name="upload_date")
    
    # AI Service Metadata
    extraction_status = Column(String(50), default="PENDING")
    global_summary = Column(Text, nullable=True)
    mind_map_plantuml = Column(Text, nullable=True)

File: Backend/docmindAI/src/main/java/com/example/docmindAI/model/Document.java
package com.example.docmindAI.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "documents")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Document {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String userId;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, name = "doc_id", unique = true)
    private String docId; // The unique ID assigned by AI-Service (e.g., timestamp)

    @Column(name = "chunk_count")
    private Integer chunkCount;

    @Column(name = "extraction_status")
    @Builder.Default
    private String extractionStatus = "PENDING";

    @Column(name = "global_summary", columnDefinition = "TEXT")
    private String globalSummary;

    @Column(name = "mind_map_plantuml", columnDefinition = "TEXT")
    private String mindMapPlantuml;

    @Column(name = "upload_date")
    private LocalDateTime uploadDate;

    @PrePersist
    protected void onCreate() {
        this.uploadDate = LocalDateTime.now();
    }
}
